Control/Monad/Logic/Class.hs

```diff
@@ -165,26 +181,109 @@ class (Monad m, Alternative m) => MonadLogic m where
     (>>-)      :: m a -> (a -> m b) -> m b
     infixl 1 >>-
 
-    -- | Logical __conditional.__ The equivalent of
-    --   <http://lpn.swi-prolog.org/lpnpage.php?pagetype=html&pageid=lpn-htmlse44 Prolog's soft-cut>.
-    --   If its first argument succeeds at all, then the results will be fed into
-    --   the success branch. Otherwise, the failure branch is taken.
-    --   satisfies the following laws:
-    --
-    --   > ifte (pure a) th el       == th a
-    --   > ifte empty th el          == el
-    --   > ifte (pure a <|> m) th el == th a <|> (m >>= th)
-    ifte       :: m a -> (a -> m b) -> m b -> m b
-
     -- | __Pruning.__ Selects one result out of many. Useful for when multiple
     --   results of a computation will be equivalent, or should be treated as
     --   such.
+    --
+    --   As an example, here's a way to determine if a number has an
+    --   factors (whole number divisors):
```

Comment:
Nice choice of an example!
I'd write "if a number is [composite](https://en.wikipedia.org/wiki/Composite_number) (has non-trivial integer divisors)" and use "composite" instead of "non-prime" below.